crates/noirc_evaluator/src/ssa/inline.rs

```diff
@@ -322,3 +329,43 @@ fn push_instruction(
     stack_frame.push(new_id);
     inline_map.insert(old_id, new_id);
 }
+
+impl node::Operation {
+    pub fn map_values_for_inlining(
+        &mut self,
+        ctx: &mut SsaContext,
+        inline_map: &HashMap<NodeId, NodeId>,
+        stack_frame: &StackFrame,
+        block_id: BlockId,
+    ) {
+        match self {
+            Operation::Binary(_) => {
+                self.map_id_mut(|id| {
+                    if let Some(a) = Memory::deref(ctx, id) {
```

Comment:
Other operations that use arrays require special treatment (store,load, call)
Binary operations that use arrays as argument all handle them the same way (and will most likely always do it)

Reply:
So it is not that binary operations deserve special treatment but rather this optimization is disabled for store, load, call instructions because of other treatment they require?
In that case it may be clearer to match on those special instructions specifically then use this optimizing function in the default case.

Reply:
It's done